src/BarracksServer/Database/BarracksDb.cs

```diff
@@ -571,5 +573,85 @@ public void SaveItem(Character character, long itemId)
 				trans.Commit();
 			}
 		}
+
+		/// <summary>
+		/// Returns all companions on given account.
+		/// </summary>
+		/// <param name="accountId"></param>
+		/// <returns></returns>
+		public List<Companion> GetCompanions(long accountId)
+		{
+			var result = new List<Companion>();
+
+			using (var conn = this.GetConnection())
+			{
+				using (var mc = new MySqlCommand("SELECT * FROM `companions` WHERE `accountId` = @accountId ORDER BY `slot`", conn))
+				{
+					mc.Parameters.AddWithValue("@accountId", accountId);
+
+					using (var reader = mc.ExecuteReader())
+					{
+						while (reader.Read())
+						{
+							var characterId = reader.IsDBNull(2) ? 0 : reader.GetInt64("characterId");
+							var companion = new Companion(reader.GetInt64("companionId"), reader.GetInt64("accountId"), characterId);
+							companion.MonsterId = reader.GetInt32("monsterId");
+							companion.Name = reader.GetStringSafe("name");
+							companion.Index = (byte)reader.GetInt32("slot");
+							companion.BarracksLayer = reader.GetInt32("barrackLayer");
+
+							var bx = reader.GetFloat("bx");
+							var by = reader.GetFloat("by");
+							var bz = reader.GetFloat("bz");
+							companion.BarracksPosition = new Position(bx, by, bz);
+
+							result.Add(companion);
+						}
+					}
+				}
+			}
+
+			return result;
+		}
+
+		/// <summary>
+		/// Set the current character associated with a companion
+		/// </summary>
+		/// <param name="companionId"></param>
+		/// <param name="characterId"></param>
+		public void SetCompanionCharacter(long companionId, long characterId)
+		{
+			using (var conn = this.GetConnection())
+			using (var trans = conn.BeginTransaction())
+			{
+				using (var cmd = new UpdateCommand("UPDATE `companions` SET {0} WHERE `companionId` = @companionId", conn, trans))
+				{
+					cmd.AddParameter("@companionId", companionId);
+					if (characterId > 0)
+						cmd.Set("characterId", characterId);
+					else
+						cmd.Set("characterId", null);
+
+					cmd.Execute();
+				}
+				trans.Commit();
+			}
+		}
+
+		/// <summary>
+		/// Deletes a companion.
```

Comment:
While the purpose of this return value can be guessed, potentially ambiguous values should be clarified.